### ISSUE TITLE
Running the quoted command with the quoted arguments

### ISSUE THREAD

**Issue by jaraco** [xonsh/xonsh#4469]
The tutorial says:

> While all Python code is also xonsh, not all Bash code can be used in xonsh. That would defeat the purpose, and Python is better anyway! Still, xonsh is Bash-wards compatible in the ways that matter, such as for running commands, reading in the Bash environment, and utilizing Bash tab completion.

Today I discovered a case where "running commands" isn't Bash-wards compatible. In an [issue I reported to PyCharm](https://youtrack.jetbrains.com/issue/PY-50694), PyCharm attempts to load the environment from the user's shell, but it fails when the user's shell is xonsh. PyCharm attempts to run the command equivalent to:

```
$ /Users/jaraco/.local/bin/xonsh -l -i -c "'/Applications/PyCharm.app/Contents/bin/printenv' 'foo'"
'/Applications/PyCharm.app/Contents/bin/printenvfoo'
```

I presume they're putting printenv and foo in single quotes because either filename could potentially have spaces in the name.

I've advised them that the uncaptured subprocess syntax (`$[]`) may be necessary for xonsh and requested they infer the presence of xonsh to apply that syntax, but I wanted to report the issue here too in case there's something that I've missed or something that xonsh could do to support this fairly simple use-case that appears to violate the shell compatibility goal of xonsh.

Is there a Bash-compatible syntax that would work in this scenario?

**Comment by anki-code** [comment]
In the nutshell the problem is that the command:
```python
'echo' '123'
```
returns string `'echo123'` instead of interpret the line as a command with two arguments and run `echo 123`.

@gforsyth @daniel-shimon @laloch what do you think about parse this kind of commands as a list of arguments?

**Comment by jaraco** [comment]
I like the idea of parsing the syntax as a list of arguments. I’ve frequently stumbled using xonsh to invoke a command with spaces in the name. The trickiest edge case might be the simplest:

```
‘cat’
```

Should that invoke the command cat or print a simple string? I’d argue if the above proposal is accepted, it should invoke the command cat.

**Comment by jaraco** [comment]
I'm just checking in on this again. It still affects me and upstream, PyCharm is awaiting a call on this issue before taking any special action to support xonsh.

**Comment by anki-code** [comment]
@jaraco I think the best way is to add explicit xonsh support like you suggested.

**Comment by gforsyth** [comment]
Yeah, we aren't going to implicitly destring things -- this is the "correct" Pythonic behavior so it's better to explicitly add xonsh support upstream.  Sorry for the delay @jaraco

**Comment by jaraco** [comment]
Respectfully, I think xonsh is making the wrong choice here, preferring an obscure Pythonism (two adjacent strings on the same line) to a common shell usage. After all xonsh exists on the premise that it will provide both syntaxes and make the hard trade offs judiciously.

Regardless, if the decision is final, would you close this issue accordingly?

**Comment by gforsyth** [comment]
Hey @jaraco -- it's generally easier for us to hold the line on "python always wins" but we've made exceptions before.  I'm not opposed but I also don't know if I have the mental bandwidth to think about the wacky edgecases that always arise when we made exceptions

**Comment by gforsyth** [comment]
I think that the case of 
```
> 'cat'
```
needs to continue returning a string `'cat'` -- either that or we check for an executable or alias named `cat` return a string if not found.

I'm much more amenable (now that I'm looking at it again) to destringing a sequence of delimited strings (I think we need to restrict this to space delimited, but that may be overly restrictive)

**Comment by jaraco** [comment]
I agree. The degenerate case of `'cat'` was perhaps a bad example and over-complicated the situation. I was leaning toward interpreting that as a command because it seems unlikely to me that someone would write a literal string and intend to see its repr, so I'm okay with only supporting the far less common `'cat' 'something'`.

**Comment by jaraco** [comment]
Good news - due to the open-source nature of VS Code, I was able to contribute native, first-class support for the xonsh shell in their startup. Starting with the 1.86 release (currently insider only), it will now extract environment variables from xonsh. And more importantly, instead of requiring special handling to emit the values as JSON, as other shells do, it was able to directly emit the environment as JSON, a more elegant solution than for other shells.

I'm still working on the PyCharm side.

**Comment by jaraco** [comment]
PyCharm now includes support for xonsh as the default shell (as of 2025.3).

**Comment by jaraco** [comment]
I reviewed the docs and I don't feel like a section on PyCharm would make sense. If you think there should be one, feel free to reopen.

**Comment by anki-code** [comment]
Thanks @jaraco !

**Comment by anki-code** [comment]
@jaraco it will be cool if you can leave the instruction with screenshot of how to setup xonsh in PyCharm.
